### ISSUE TITLE
drawer modality curtain not working

### ISSUE THREAD

**Issue by akabanov57** [vaadin/flow-components#3288]
### Description

When drawer is opened in full screen, the content page is not moved to the right side instead, the drawer is overlaid on the page with the background transparent. In this case, I expect the controls in that area to not respond to on click. But this is not true for small devices with touch screen. On desktop it works as expected.

### Expected outcome

I expect the controls in that area to not respond to on click.

### Minimal reproducible example

```java
public class MainLayout extends AppLayout {
  public MainLayout() {
    DrawerToggle toggle = new DrawerToggle();

    H1 title = new H1("MyApp");
    title.getStyle().set("font-size", "var(--lumo-font-size-l)").set("margin", "0");

    Tabs tabs = getTabs();

    addToDrawer(tabs);

    VerticalLayout btnLayout = new VerticalLayout();
    Button testBtn = new Button("Test");
    testBtn.addClickListener(
        e -> {
          Notification.show("Test clicked.");
        });
    btnLayout.add(testBtn);
    btnLayout.setDefaultHorizontalComponentAlignment(Alignment.END);

    addToNavbar(toggle, title, btnLayout);
    setPrimarySection(Section.DRAWER);
  }

  private Tabs getTabs() {
    Tabs tabs = new Tabs();
    tabs.add(
        createTab(VaadinIcon.DASHBOARD, "Dashboard"),
        createTab(VaadinIcon.CART, "Orders"),
        createTab(VaadinIcon.USER_HEART, "Customers"),
        createTab(VaadinIcon.PACKAGE, "Products"),
        createTab(VaadinIcon.RECORDS, "Documents"),
        createTab(VaadinIcon.LIST, "Tasks"),
        createTab(VaadinIcon.CHART, "Analytics"));
    tabs.setOrientation(Tabs.Orientation.VERTICAL);
    return tabs;
  }

  private Tab createTab(VaadinIcon viewIcon, String viewName) {
    Icon icon = viewIcon.create();
    icon.getStyle()
        .set("box-sizing", "border-box")
        .set("margin-inline-end", "var(--lumo-space-m)")
        .set("margin-inline-start", "var(--lumo-space-xs)")
        .set("padding", "var(--lumo-space-xs)");

    RouterLink link = new RouterLink();
    link.add(icon, new Span(viewName));
    link.setTabIndex(-1);

    return new Tab(link);
  }
}
```

### Steps to reproduce

On desktop.
1. Run snippet and go to localhost:8080.
2. Make browser window small. Menu should disappear.
3. Click drawer toggle. Now click Test Button. Menu disappear and there is no message from button.

Now do the same on smartphone.
You will see message "Test button clicked."

### Environment

Vaadin version(s): 23.0.10
OS: Ubuntu 20.04

### Browsers

Chrome, Firefox

**Comment by rolfsmeds** [comment]
I was able to reproduce this in V23.0.10, but not in V23 beta1, so this seems to be a regression that has occurred somewhere between those.

**Comment by web-padawan** [comment]
I could not reproduce this issue in Vaadin 22, but it exists in both Vaadin 23 and Vaadin 23.1 versions.
Note: it can be reproduce also in Chrome DevTools mobile device emulator, see the recording:

https://user-images.githubusercontent.com/10589913/172575300-d22b6f5a-f6b1-4e46-9bb0-72431423defe.mp4

It seems like the regression is related to removing the "ghost click" prevention done in #3169

**Comment by akabanov57** [comment]
vaadin 23.1.0. Still not working.

**Comment by web-padawan** [comment]
The fix for this issue will be included to the next patch release (23.1.1)
